### PR TITLE
kem: Allows registering and querying by CurveID

### PR DESCRIPTION
**Pull request by bwesterb**

### CODE REVIEW

File: kem/hybrid/hybrid.go [hunk 1]
```diff
@@ -56,18 +56,21 @@ var kyber512X kem.Scheme = &scheme{
 	"Kyber512-X25519",
 	kyber512.Scheme(),
 	hpke.KEM_X25519_HKDF_SHA256.Scheme(),
+	0x01fd,
```
Comment (bwesterb) [inline kem/hybrid/hybrid.go:59]: These are temporary and we might not want to include them, yet.

Comment (armfazh) [reply kem/hybrid/hybrid.go:59]: from where these numbers came from, please add a link to a reference.

File: kem/tls/tls.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Package tls adds helpers to integrate the KEMs supported in Circl into TLS.
```
Comment (armfazh) [inline kem/tls/tls.go:1]:
```suggestion
// Package tls adds helpers to integrate the KEMs supported in CIRCL into TLS.
```

**Comment by armfazh** [comment]
Looks good, but CI is not happy.

File: kem/tls/tls.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	allSchemesByCurveID = make(map[uint]kem.Scheme)
+	for _, scheme := range schemes.All() {
+		if tlsScheme, ok := scheme.(TLSScheme); ok {
+			allSchemesByCurveID[tlsScheme.TLSCurveID()] = scheme
+		}
+	}
+}
+
+// Returns scheme with the given TLS CurveID, if any.
+func SchemeByCurveID(id uint) kem.Scheme { return allSchemesByCurveID[id] }
```
Comment (armfazh) [inline kem/tls/tls.go:23]: this function may be moved to `/circl/kem/schemes` which is the place where the register lives, doing so we don't need the `/circl/kem/tls` package, which may be confused with the stdlib package.

Comment (bwesterb) [reply kem/tls/tls.go:23]: I used a separate package to be consistent with the pki package, that provides similar functionality but then querying signature schemes by OID. If you’re ok being inconsistent with that, I’ll move it.

Comment (armfazh) [reply kem/tls/tls.go:23]: the issue I'm pointing is that when importing `/circl/kem/tls`, it looks like is a replacement of the `tls` package. what about renaming to `tls_helpers`?
or putting these functions in the `pki` package.

**Comment by armfazh** [comment]
@bwesterb  Let's rebase this PR if it's still needed.
CurveIDs may have changed since then.
cc: @Lekensteyn

**Comment by bwesterb** [comment]
Yeah, let's just start fresh with this if we need it.

**Comment by bwesterb** [comment]
(Closing for now.)

**Pull request closed without merging**